### PR TITLE
fix(trainer): require explicit classification lifecycle values

### PR DESCRIPTION
**Pull request by liq22**
## Objective

Close the verified Trainer lifecycle default split for maintained classification Pipelines without materializing Schema defaults or introducing a second runtime mapping.

## Proven invariants

```text
trainer.num_epochs
= one visible positive integer authority
= value passed to Lightning max_epochs
```

For maintained P01/P02/P05 classification lifecycles:

```text
trainer.test_after_fit
= one visible boolean policy
= exact Runtime decision to run or skip post-fit evaluation
```

## Core implementation

- require `trainer.num_epochs` at the complete strict Schema boundary;
- reject legacy `trainer.max_epochs` as a second epoch authority;
- make `Default_trainer` consume the exact visible integer without alias, type-conversion, or missing-value fallback;
- require `trainer.test_after_fit` only for P01/P02/P05 classification lifecycles;
- remove classification Runtime's hidden `test_after_fit=True` fallback;
- preserve explicit `test_after_fit=false` as a training-only decision rather than truthiness-coercing it to evaluation.

## Pipeline-aware configuration migration

`default_single_gpu.yaml` is shared by classification and generative Pipelines, so this PR deliberately does **not** place `test_after_fit` in that shared base.

The policy is declared at each maintained classification experiment layer, including:

- all validated classification demos;
- the MFPT transparent baseline;
- the P07 executable operator-path smoke.

Pipeline 06 remains free of the unused classification field.

## Actual changed-file scope

The PR changes **22 files**:

```text
Core semantic code:                 3
  src/config_schema/models.py
  src/runtime/classification.py
  src/trainer_factory/Default_trainer.py

Maintained experiment configs:     11
  9 classification demos
  1 MFPT baseline
  1 P07 executable smoke

Focused tests:                      5
  test/test_classification_runtime.py
  test/test_config_analysis_parity.py
  test/test_phmfactory_config_resolver.py
  test/test_trainer_lifecycle_contract.py
  test/test_trainer_lifecycle_schema.py

Documentation/workflows:            3
  configs/base/trainer/README.md
  .github/workflows/core-quality-gates.yml
  .github/workflows/mfpt-baseline.yml
```

The category counts sum to the exact 22-file Git diff. The MFPT workflow change only installs the already-declared public configuration dependency `pydantic`.

## User-visible behavior

Evaluated classification:

```yaml
trainer:
  num_epochs: 10
  test_after_fit: true
```

Training-only classification:

```yaml
trainer:
  num_epochs: 10
  test_after_fit: false
```

The following now fail before Pipeline import on the public path:

```yaml
trainer: {test_after_fit: true}             # missing num_epochs
trainer: {num_epochs: 10}                   # classification policy missing
trainer: {num_epochs: "10", test_after_fit: true}
trainer: {num_epochs: 10, max_epochs: 20, test_after_fit: true}
trainer: {num_epochs: 10, test_after_fit: "false"}
```

Direct/legacy classification and Trainer callers are independently guarded so they cannot recover deleted defaults after bypassing the public Schema boundary.

## Boundaries

- no device/count migration in this PR;
- no scheduler, monitor, early-stopping, pruning, log-cadence, checkpoint-selection, result-root, metric, split, model, objective, or estimator change;
- the MFPT algorithm and protocol are unchanged; only its evaluation policy is made visible and the workflow receives the required Schema dependency;
- no P06 lifecycle redesign;
- no Schema-default materialization or `model_dump()` runtime replacement;
- no manager, wrapper, registry, manifest, hash, ledger, receipt, retry, or fallback.

## Validation completed

All triggered checks passed on head `18be684ea4c76246c18760264fa7cdd2f32a7dac`:

```text
PHMFactory public package       PASS
Core quality gates              PASS
MFPT real-data baseline         PASS
PHMFactory release readiness    PASS
Repository layout               PASS
Submodule policy                PASS
```

The evidence includes:

- strict Schema/entrypoint lifecycle counterexamples;
- direct Runtime and Trainer counterexamples;
- clean-wheel installation and public command execution;
- complete offline Dummy fit → checkpoint restore → test → aggregate lifecycle;
- MFPT pinned-data three-seed CPU run, split verification, finite metrics, checkpoints, and aggregation;
- Pipeline 06 and UXFD regression contracts.

## Status after merge

```text
Q0b-A  explicit environment execution contract     -> CLOSED
Q0b-B1 explicit classification Trainer lifecycle   -> CLOSED
Q0b-B2 explicit device/count authority              -> VERIFIED OPEN
Q0b-B3 remaining callback/logging behavior defaults -> VERIFIED OPEN
```

## Rollback

Revert the single squash commit.

### CODE REVIEW

**Comment by liq22** [comment]
Follow-up convergence note: PR #211's explicit `num_epochs`/`test_after_fit` contract remains closed. The next bounded device-count PR will not modify epoch or post-fit evaluation semantics.
